Cadasta/cadasta-test
# Convert to a package, use pytest, debug tests, and do some clean-up

### Proposed changes in this pull request
- Convert into a Python package:
    - Add **setup.py**.
    - Move main files and directories into the `cadasta.test` module (following the convention laid out in the cadasta-sdk repository) and add **\_\_init\_\_.py** files as needed.
    - Rename directories containing test files to **\*_test**.
    - Rename **resources** directory to **files** so that test files that are not resources can be added here.
- Use pytest:
    - Delete **cadasta_test_suites.py** because tests should be automatically discoverable instead of manually adding them to a list. Note: tests that were commented out in this file are also made undiscoverable from pytest by renaming the test "xtest\_\*" instead of "test\_\*".
    - Replace **runtests** shell script with **runtests.py** Python script.
    - Add argparse to **runtests.py** and update arguments of **runtests.py** to allow passing optional arguments to pytest and to allow specifying the Selenium WebDriver to use (Chrome by default, and Firefox as an alternative). Update **README.md** to match.
- Debug some tests:
    - With the proper test fixtures, all account tests should now pass.
    - Remove references to and tests for the second password field that no longer exists in the UI.
    - Update error messages.
    - Avoid updating the entity data.
- Do some more clean-up:
    - Make internal import paths relative instead of absolute.
    - Move the `setUp()` and `tearDown()` methods from the child classes to the parent `SeleniumTestCase` class, and add `super()` calls as needed.

### When should this PR be merged
When convenient.

### Risks
None foreseen.

### Follow-up actions
None (yet).

### Checklist (for reviewing)

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description. 
- [ ] **Is the PR labeled correctly?** It should have the `migration` label if a new migration is added. 
- [ ] **Is the risk level assessment sufficient?** The risks section should contain all risks that might be introduced with the PR and which actions we need to take to mitigate these risks. Possible risks are database migrations, new libraries that need to be installed or changes to deployment scripts.

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components. 

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**
